common/src/primitives/id.rs

```diff
@@ -42,12 +42,26 @@ impl From<Uint256> for H256 {
     }
 }
 
-#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Encode, Decode)]
+#[derive(Debug, Clone, PartialEq, Eq, Encode, Decode)]
 pub struct Id<T: ?Sized> {
     id: H256,
     _shadow: std::marker::PhantomData<T>,
 }
 
+// We implement Ord manually to avoid it getting inherited to T through PhantomData, because Id having Ord doesn't mean T requiring Ord
+impl<T: Eq> Ord for Id<T> {
```

Comment:
What about also implementing `Eq` and `PartialEq` for `Id<T>` and removing the `T: Eq` requirement here?

Reply:
Well, something that isn't Eq can't be Idable. This makes sense. 

If you're happy with this answer, please resolve this discussion.